src/parser.js

```diff
@@ -606,6 +606,11 @@ define([
             if (mug.options.isControlOnly) {
                 // HACK fix abstraction broken by direct tree insert
                 form.mugMap[mug.ufid] = mug;
+            } else if (form.getAbsolutePath(mug) &&
+                       form.getAbsolutePath(mug) !== form.getControlPath(mug)) {
+                var parentPath = form.getAbsolutePath(mug).split('/');
```

Comment:
Splitting the path might be problematic since removing the last element of a mug's path is not guaranteed to get the parent mug's path (this is at least true for model iteration Repeat questions). Would be better to get the parent path like this:
```js
// might need to check if the node is null/undefined before traversing to .parent
parentNode = form.dataTree.getNodeFromMug(mug).parent
mug.p.dataParent = parentNode.getAbsolutePath()
```
This is going to require a refactor of `Tree.getAbsolutePath`: move/rename [`pathOf`](https://github.com/dimagi/Vellum/blob/non-group-child/src/tree.js#L279) to `Node.getAbsolutePath()` so you can call `parent.getAbsolutePath()` even if parent happens to be the root node.